float64 v
float64 w
float64 dt
float64 v_tot
float64 w_tot
